### ISSUE TITLE
Inkscape executable not found on Windows when not associated with svg file type

### ISSUE THREAD

**Issue by jasongullifer** [jupyter/nbconvert#456]
Hi,

I was having an issue converting a jupyter notebook to pdf. I kept getting the error "Inkscape executable not found" (from nbconvert/nbconvert/preprocessors/svg2pdf.py). After looking though the code, I realized it was grabbing the path from the windows registry. However there were at least 2 problems for me:

1) inkscape doesn't associate svg files with inkscape.exe (so the path wouldn't exist in the registry yet) right away after installation

2) after I manually associated svg files with inkscape, the association is added to the registry, but it added to a different key than svg2pdf.py expects

To expand on (2):
svg2pdf tries to get the path from the registry key "HKEY_LOCAL_MACHINE\\SOFTWARE\\Classes\\inkscape.svg\\DefaultIcon" which did not exist after I made the association in the Windows "open with" menu. 

However, the following key does exist: "HKEY_LOCAL_MACHINE\SOFTWARE\Classes\svgfile\shell\Inkscape\command" and points to inkscape.

I solved this by adding the expected key to the registry, but I think the code should be updated.

Thanks!

Let me know i f you need any more information.

**Comment by takluyver** [comment]
Does that key exist after a standard installation of Inkscape? Ideally it should be able to find it regardless of the file association.

**Comment by takluyver** [comment]
@michaelpacer there may still be something for us to do here. It's 'solved' in that Jason made things work on his computer, but we might still want to change something if other people are likely to run into this in the future. Was there a specific reason you closed it?

**Comment by michaelpacer** [comment]
Trying to clean up issues that were dealt with, I misread this and thought it had been addressed.

**Comment by michaelpacer** [comment]
Also, I don't know if we explicitly declare a dependency on Inkscape or give standard instructions on how to install it.

BTW if it is not getting the correct path, then this PR probably relates and points out the code that needs updating: https://github.com/jupyter/nbconvert/pull/131

**Comment by takluyver** [comment]
No problem. I've retitled it to be clearer. Yep, #131 is where the relevant code was added.

**Comment by michaelpacer** [comment]
Do we know why this might be different in different versions of windows?

**Comment by juhasch** [comment]
I tried this on three different machines, and Inkscape always creates the "HKEY_LOCAL_MACHINE\SOFTWARE\Classes\inkscape.svg\DefaultIcon" entry. One could argue if 
"HKEY_LOCAL_MACHINE\SOFTWARE\Classes\inkscape.svg\shell\edit\command" would be the better option here, but this is another matter.

On the machines I tested, this registry entry does not exist:
"HKEY_LOCAL_MACHINE\SOFTWARE\Classes\svgfile\shell\Inkscape\command"
 
@jasongullifer What Inkscape and Windows version are you using ?

**Comment by michaelpacer** [comment]
I'm going to close this and if we discover that it is a more general problem that we can address, we can reopen.

**Comment by galckmin** [comment]
I had the same problem

**Comment by ManPython** [comment]
Problem still exist.


500 : Internal Server Error

The error was:

nbconvert failed: Inkscape executable not found



Just take some example, eg. 
https://github.com/odewahn/ipynb-examples/blob/master/Part%205%20-%20Rich%20Display%20System.ipynb

**Comment by dileep3004** [comment]
![screenshot 47](https://user-images.githubusercontent.com/17436236/47239735-bf0b9d80-d403-11e8-80db-664ee981dad2.png)

**Comment by dileep3004** [comment]
How to Resolve the Above Error

**Comment by JustxYHG** [comment]
> How to Resolve the Above Error

Have you resolved this error? I have the same problem

**Comment by ManPython** [comment]
I used commercial Wolfram Mathematica : )
https://reference.wolfram.com/language/howto/ExportToPDF.html

The nature of Open Source in case of bugs is horrible life with problems.
Should exist some "fund" to solve top bugs of OS software :D

**Comment by juhasch** [comment]
You can also manually set where Jupyter will look for the inkscape executable in the configuration files.
Steps:
1. Find out where your inkscape executable is located,
 e.g. `C:\\Program Files\\Inkscape\\inkscape.exe` for me.
2. Find your Jupyter configuration files using `jupyter --path`
  They are called `jupyter_nbconvert_config.py` and `jupyter_notebook_config.py`
3. If these configuration files don't exist, create them using `jupyter notebook --generate-config`
4. Add the line `c.SVG2PDFPreprocessor.inkscape = '"C:\\Program Files\\Inkscape\\inkscape.exe"'` (put in the correct path for your installation here)
 to your configuration files. It is necessary to add them to **both** files so it will work with the command line tool `nbconvert` and in the browser.

**Comment by rbeesley** [comment]
I added this to #139:

On Windows, the code looks in the registry for this key: `HKEY_LOCAL_MACHINE\SOFTWARE\Classes\inkscape.svg\DefaultIcon`. This key likely would have a REG_SZ string which is just the path to inkscape.exe. I was able to fix this for myself by creating this key and setting the value to where inkscape is installed. Long term, `Lib\site-packages\nbconvert\preprocessors\svg2pdf.py` could be updated to search the path for the executable. Python 3.3 has [shutil.which()](https://docs.python.org/dev/library/shutil.html#shutil.which), which could do this and appears independent of the platform.

To me, this seems like the best approach for Python 3.3 and higher, which is the base version required of Python 3.x installs. For Python 2.7 a replacement method could be written and/or if inkscape is not in the PATH, it could fall back to the current method. This would resolve the current problem for many users. Python 2.7 is approaching EOL, so I think the replacement for shutil.which on Python 2.7 could be safely skipped assuming support will be dropped in the next few months.
